resip/recon/UserAgentClientPublication.cxx

```diff
@@ -0,0 +1,114 @@
+#include "UserAgent.hxx"
+#include "ReconSubsystem.hxx"
+#include "UserAgentClientPublication.hxx"
+
+#include <rutil/Log.hxx>
+#include <rutil/Logger.hxx>
+#include <resip/dum/DialogUsageManager.hxx>
+#include <resip/dum/ClientPublication.hxx>
+
+using namespace recon;
+using namespace resip;
+using namespace std;
+
+#define RESIPROCATE_SUBSYSTEM ReconSubsystem::RECON
+
+UserAgentClientPublication::UserAgentClientPublication(UserAgent& userAgent, DialogUsageManager& dum, unsigned int handle)
+: AppDialogSet(dum),
+  mUserAgent(userAgent),
+  mDum(dum),
+  mPublicationHandle(handle),
+  mEnded(false)
+{
+   mUserAgent.registerPublication(this);
+}
+
+UserAgentClientPublication::~UserAgentClientPublication()
+{
+   mUserAgent.unregisterPublication(this);
+}
+
+PublicationHandle 
+UserAgentClientPublication::getPublicationHandle()
+{
+   return mPublicationHandle;
+}
+
+void 
+UserAgentClientPublication::end()
+{
+   if(!mEnded)
+   {
+      mEnded = true;
+      AppDialogSet::end();
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// ClientPublicationHandler ///////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+void
+UserAgentClientPublication::onSuccess(ClientPublicationHandle handle, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onSuccess - not implemented\n");
+}
+
+void
+UserAgentClientPublication::onRemove(ClientPublicationHandle handle, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onRemove - not implemented\n");
+}
+
+void
+UserAgentClientPublication::onFailure(ClientPublicationHandle handle, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onFailure - not implemented\n");
+}
+
+int
+UserAgentClientPublication::onRequestRetry(ClientPublicationHandle handle, int retrySeconds, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onRequestRetry - not implemented\n");
+   return 30;
+}
+
+void
+UserAgentClientPublication::onStaleUpdate(ClientPublicationHandle handle, const SipMessage& status)
+{
+   InfoLog(<<"UserAgentClientPublication::onStaleUpdate - not implemented\n");
+}
+
+/* ====================================================================
+
+ Copyright (c) 2007-2008, Plantronics, Inc.
+ Mateus Bellomo (mateusbellomo AT gmail DOT com) https://mateusbellomo.wordpress.com/
```

Comment:
@MateusBellomo please start this line with "Copyright (C) 2016, " before your name, same for the hxx file